bundles/ui/org.eclipse.smarthome.ui.icon/src/main/java/org/eclipse/smarthome/ui/icon/IconProvider.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.ui.icon;
+
+import java.io.InputStream;
+import java.util.Locale;
+import java.util.Set;
+
+import org.eclipse.smarthome.ui.icon.IconSet.Format;
+
+/**
+ * An icon provider can provide {@link InputStream}s for icons.
+ * The source of the images can depend on the provider implementation.
+ * The byte stream represents a PNG or SVG image, depending on the format.
+ *
+ * The icon category corresponds to the list of available channel categories.
+ *
+ * In order to provide means to user interfaces to know, what kind of icon sets are available in the system (and offered
+ * by some icon provider), the provider can additionally provide a set of {@link IconSet}s.
+ *
+ * @author Kai Kreuzer - Initial contribution
+ */
+public interface IconProvider {
+
+    /**
+     * Returns a set of icon set definitions (meta-data) which this provider defines.
+     *
+     * @return a set of icon sets in the default locale
+     */
+    Set<IconSet> getIconSets();
```

Comment:
Does one IconProvider really needs to return multiple IconSets?

Otherwise we could name the interface IconSetProvider.